javascripts/discourse/components/toc-contents.gjs

```diff
@@ -71,17 +80,27 @@ export default class TocContents extends Component {
       return;
     }
 
-    this.headingPositions = Array.from(headings).map((heading) => {
-      const id = this.getIdFromHeading(heading);
-      return {
-        id,
-        position:
-          heading.getBoundingClientRect().top +
-          window.scrollY -
-          headerOffset() -
-          POSITION_BUFFER,
-      };
-    });
+    const sameIdCount = new Map();
+    const mappedToc = this.mappedToc;
+    this.headingPositions = Array.from(headings)
+      .map((heading) => {
+        const id = this.tocProcessor.getIdFromHeading(
+          this.args.postID,
+          heading,
+          sameIdCount
+        );
+        return mappedToc[id]
```

Comment:
To explain why I rewrote this: This is a very small bug I found during debugging, if there are nodes like `blockquote > h2`, we need to remove them because `tocStructure` does not contain them. They will interfere with `activeHeading`.